src/Sulu/Bundle/ActivityBundle/Infrastructure/Doctrine/Repository/ActivityRepository.php

```diff
@@ -102,8 +102,11 @@ protected function getInsertQueryBuilder(ActivityInterface $activity): QueryBuil
         }
 
         // set value of id explicitly if class has a pre-insert identity-generator to be compatible with postgresql
-        if (!$classMetadata->idGenerator->isPostInsertGenerator()) {
-            $queryBuilder->setValue($classMetadata->getColumnName('id'), $classMetadata->idGenerator->generate($this->entityManager, $activity));
+        if (
+            null !== $classMetadata->idGenerator
+            && !$classMetadata->idGenerator->isPostInsertGenerator()
+        ) {
+            $queryBuilder->setValue($classMetadata->getColumnName('id'), $classMetadata->idGenerator->generateId($this->entityManager, $activity));
```

Comment:
updated this to use the newer generator and as the idGenerator can be null to check for that also.